src/libs/actions/TaxRate.ts

```diff
@@ -119,10 +182,299 @@ function clearTaxRateError(policyID: string, taxID: string, pendingAction?: Pend
+                            acc[taxID] = {
+                                isDisabled: !!originalTaxes[taxID].isDisabled,
+                                pendingFields: {isDisabled: null},
+                                errorFields: {isDisabled: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.updateFailureMessage')},
+                            };
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        taxFieldsArray: JSON.stringify(taxesIDsToUpdate.map((taxID) => ({taxCode: taxID, enabled: isEnabled}))),
+    } satisfies SetPolicyTaxesEnabledParams;
+
+    API.write(WRITE_COMMANDS.SET_POLICY_TAXES_ENABLED, parameters, onyxData);
+}
+
+type TaxRateDeleteMap = Record<
+    string,
+    | (Pick<TaxRate, 'pendingAction'> & {
+          errors: OnyxCommon.Errors | null;
+      })
+    | null
+>;
+
+/**
+ * API call to delete policy taxes
+ * @param taxesToDelete A tax IDs array to delete
+ */
+function deletePolicyTaxes(policyID: string, taxesToDelete: string[]) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const policyTaxRates = policy?.taxRates?.taxes;
+
+    if (!policyTaxRates) {
+        throw new Error('Policy or tax rates not found');
+    }
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesToDelete.reduce<TaxRateDeleteMap>((acc, taxID) => {
+                            acc[taxID] = {pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE, errors: null};
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesToDelete.reduce<TaxRateDeleteMap>((acc, taxID) => {
+                            acc[taxID] = null;
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesToDelete.reduce<TaxRateDeleteMap>((acc, taxID) => {
+                            acc[taxID] = {
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
```

Comment:
We should clear the `pendingAction` on delete case
```suggestion
                                pendingAction: null
```